doc/developer-guide/api/functions/TSUrlStringGet.en.rst

```diff
@@ -61,6 +62,29 @@ terminator - :arg:`length` must be used to correctly display the string.
 This function is useful to guarantee a URL that is as complete as possible given the specific
 request.
 
+:func:`TSHttpHdrEffectiveUrlBuffGet` returns the effective URL for any HTTP request (not just the client request).
+If the request has a Host header field, the host specifier it provides is inserted into the URL.  The host and
+scheme in the returned URL will be normalized to lower case letters (to make URL comparisons simple and fast).
+This prints the effective URL for the header specified by :arg:`hdr_buf` and
+:arg:`hdr_loc` to the buffer starting at :arg:`buff`. If the effective URL is longer than :arg:`size`, nothing is
+written to :arg:`buff`.  Note that this is not considered an error case, the function will still reeturn `TS_SUCCESS`.
+The full length of the URL is always returned in :arg:`*length` when the function returns `TS_SUCCESS`.
```

Comment:
"It is the responsibility of the caller to check this result to determine if output was generated."